cmd/agent.go

```diff
@@ -345,6 +348,33 @@ func (ar *AgentRunner) runInstance() error {
 
 		logger.Debug("Running plugin", "source", source)
 
+		uuidSeedData := map[string]string{
+			// Repeatably identify this plugin / policy combination.
+			// Rerunning this plugin / policy on the same agent with the same config should generate the same UUID.
+			"plugin-name":    pluginName,
+			"plugin-version": "1.0.0",
+			"policy-version": "v1.2.3",
+
+			// Uniquely identify this agent.
+			// If a set of machines is running the same agent config, each should have a unique UUID.
+			"agent-version": "v1.0.0",
+			"hostname":      os.Getenv("HOSTNAME"),
+		}
+
+		seed := ""
+		for k, v := range uuidSeedData {
+			seed = fmt.Sprintf("%s-%s-%s", seed, k, v)
```

Comment:
Would probably be nicer to use something like `strings.Join`: https://www.geeksforgeeks.org/strings-join-function-in-golang-with-examples/

Reply:
I've updated the implementation, as this one was still causing headaches due to ordering of the map. 
If you ran the generator 1000 times, about 40% would be incorrect because the generated strings wasn't ordered. 
New implementation also uses checksum for math reasons. 
```
"plugin:v1.3.0" and "plugin:v1.3.1" generated the same UUID -> Because it only takes into account the first 8 bytes in a byte array. In this case the n>8th byte was different. 
The new implementation gives us a better guarantee. 
```